docs/source/usage/plugins/energyHistogram.rst

```diff
@@ -3,10 +3,67 @@
 Energy Histogram
 ----------------
 
-This plugin computes the energy histogram particle species simulated with PIConGPU.
-The energy binning can be set up using command line parameters.
-A *virtual detector* can be located in the y direction of the simulation to count only particles that would reach such a detector in an experimental setup.
-If the detector setup is not specified, all particles of a species are considered in the histogram. 
+This plugin computes the energy histogram (spectrum) of a selected particle species and stores it to plain text files.
+The acceptance of particles for counting in the energy histogram can be adjusted, e.g. to model the limited acceptance of a realistic spectrometer.
+
+.param file
+^^^^^^^^^^^
+
+The :ref:`particleFilters.param <usage-params-core>` file allows to define accepted particles for the energy histogram.
+A typical :ref:`filter <usage-particles>` could select particles within a specified opening angle in forward direction.
+
+For example, to limit to particles within a cone with an opening angle of five degrees (pinhole):
+
+.. code:: cpp
+
+   namespace picongpu
+   {
+   namespace particles
+   {
+   namespace filter
+   {
+       struct FunctorParticlesForwardPinhole
+       {
+           static constexpr char const * name = "forwardPinhole";
+
+           template< typename T_Particle >
+           HDINLINE bool operator()(
+               T_Particle const & particle
+           )
+           {
+               float3_X const mom = particle[ momentum_ ];
+
+               /* place detector in y direction, "infinite distance" to target,
+                * and five degree opening angle
+                */
+               float_X const openingAngle = 5.0 * PI / 180.;
+               float_X const dotP = mom.y() / math::abs(mom);
```

Comment:
REQUIRED / QUESTION:
How do you prevent NaNs for zero momentum values?

Reply:
right, that's what we usually add. I will add an if clause and ignore resting particles!